File: app/src/main/java/com/soya/launcher/view/KeyEventFrameLayout.java
package com.soya.launcher.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.KeyEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class KeyEventFrameLayout extends MyFrameLayout {
    private KeyEventCallback keyEventCallback;

    public KeyEventFrameLayout(@NonNull Context context) {
        this(context, null);
    }

    public KeyEventFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean dispatchKeyEvent(KeyEvent event) {
        if (keyEventCallback != null) keyEventCallback.onKeyDown(event);
        return super.dispatchKeyEvent(event);
    }

    public void setKeyEventCallback(KeyEventCallback keyEventCallback) {
        this.keyEventCallback = keyEventCallback;
    }

    public interface KeyEventCallback{
        void onKeyDown(KeyEvent event);
    }
}

File: app/src/main/java/com/soya/launcher/view/KeyEventFrameLayout.kt
package com.soya.launcher.view

import android.content.Context
import android.util.AttributeSet
import android.view.KeyEvent

class KeyEventFrameLayout @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
    MyFrameLayout(context, attrs) {
    private var keyEventCallback: KeyEventCallback? = null

    override fun dispatchKeyEvent(event: KeyEvent): Boolean {
        if (keyEventCallback != null) keyEventCallback!!.onKeyDown(event)
        return super.dispatchKeyEvent(event)
    }

    fun setKeyEventCallback(keyEventCallback: KeyEventCallback?) {
        this.keyEventCallback = keyEventCallback
    }

    interface KeyEventCallback {
        fun onKeyDown(event: KeyEvent?)
    }
}
